### ISSUE TITLE
kubeflow v1.1/1.2 upstream changes to jupyter-apis component

### ISSUE THREAD

**Issue by saffaalvi** [StatCan/jupyter-apis#42]
Summary of upstream changes that we may have to integrate into the `jupyter-apis` component when upgrading to v1.1/1.2.

**Comment by saffaalvi** [comment]
Tracking PR from upstream with a summary of changes https://github.com/kubeflow/kubeflow/pull/5280/ 
## **UI Changes** ##
**Notebook Table**
- Reduced columns: got rid of the Status header and CPU, Memory, and Volumes columns completely.
- Can now interact more with notebook servers: start, stop, or delete (options menu instead of trash icon)
  * When stop/delete is selected, a dialog appears to warn user and confirm action

**Volumes Table**

While we have the volumes table underneath the notebook servers table, the new kubeflow version has it as an extra menu option in the left sidebar. This table has additional features such as:
- Displaying Access Mode and Storage Class
- Allowing user to browse volume contents: view, download, open, upload, etc.
- Creating a new volume or cloning an existing snapshot right from the volumes table (don't have to create a notebook server first)

![volumes](https://user-images.githubusercontent.com/57377830/99314054-816dde80-282e-11eb-90a6-84518763ac58.png)
![volumes2](https://user-images.githubusercontent.com/57377830/99314061-829f0b80-282e-11eb-8005-cdf3b2ce5dfd.png)

in progress...

**Comment by saffaalvi** [comment]
There is more information for what v1.2 will have in their release description https://github.com/kubeflow/kfctl/releases/tag/v1.2-rc.0 and this roadmap issue https://github.com/kubeflow/kubeflow/issues/5224.

Some features mentioned below were found from the following PRs, issues, and blog post:
Not merged: https://github.com/kubeflow/kubeflow/pull/5280/ 
Merged: https://github.com/kubeflow/kubeflow/pull/5332

## Notebooks Table
- Reduced columns: got rid of the Status header and CPU, Memory, and Volumes columns completely.
  * Make expandable rows to show this information instead (?)

**Comment by saffaalvi** [comment]
When running **MiniKF**, these are some changes I've seen that I'm not sure have been/will be implemented in v1.1 or v1.2:
## Notebooks Table 
- Can now interact more with notebook servers: start, stop, or delete (options menu instead of trash icon)
  * When stop/delete is selected, a dialog appears to warn user and confirm action

## Volumes Table 
While we have the volumes table underneath the notebook servers table, the new kubeflow version in **MiniKF** has it as an extra menu option in the left sidebar. This table has additional features such as:
- Displaying Access Mode and Storage Class
- Allowing user to browse volume contents: view, download, open, upload, etc.
- Creating a new volume or cloning an existing snapshot right from the volumes table (don't have to create a notebook server first)

![volumes](https://user-images.githubusercontent.com/57377830/99314054-816dde80-282e-11eb-90a6-84518763ac58.png)
![volumes2](https://user-images.githubusercontent.com/57377830/99314061-829f0b80-282e-11eb-8005-cdf3b2ce5dfd.png)

**Comment by ca-scribner** [comment]
@sylus @zachomedia @brendangadd @blairdrummond  ^ 

What is/isn't in 1.2 for the jupyter web app is hella confusing, but Saffa's got a solid summary if anyone needs a good place to start for figuring it out.

**Comment by saffaalvi** [comment]
## v1.1 - Manifest Tags
_FEB 6 - Delete dead code for releasing and updating images_
https://github.com/kubeflow/kubeflow/commit/d9be4b9e14fc200c2aa2140a7b0b2774ffded177
**TAG PRESENT IN:**

`jupyter/jupyter-web-app/base/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/jupyter-web-app
newName: gcr.io/kubeflow-images-public/jupyter-web-app
newTag: vmaster-gd9be4b9e

`jupyter/jupyter-web-app/base_v3/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-gd9be4b9e

`stacks/ibm/application/jupyter-web-app/base/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-gd9be4b9e

`stacks/azure/application/jupyter-web-app/base/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-gd9be4b9e

_FEB 25 - [CentralDashboard][Testing] End-to-end testing added_
https://github.com/kubeflow/kubeflow/commit/f39279c02c83b875b19c85676f327cac410713a8
**TAG PRESENT IN:**

`jupyter/notebook-controller/base/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/notebook-controller
  newName: gcr.io/kubeflow-images-public/notebook-controller
  newTag: vmaster-gf39279c0

`jupyter/notebook-controller/base_v3/kustomization.yaml`
- name: gcr.io/kubeflow-images-public/notebook-controller
  newName: gcr.io/kubeflow-images-public/notebook-controller
  newTag: vmaster-gf39279c0

**Comment by saffaalvi** [comment]
for v1.2
AUG 7 - Fix #5111: Trim whitespace from custom images
https://github.com/kubeflow/kubeflow/commit/845af298b733802e527562702cf283b6cc5b85cc
TAG PRESENT IN:
jupyter/jupyter-web-app/base/kustomization.yaml
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-g845af298
jupyter/jupyter-web-app/base_v3/kustomization.yaml
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-g845af298
stacks/ibm/application/jupyter-web-app/base/kustomization.yaml
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-g845af298
stacks/azure/application/jupyter-web-app/base/kustomization.yaml
- name: gcr.io/kubeflow-images-public/jupyter-web-app
  newName: gcr.io/kubeflow-images-public/jupyter-web-app
  newTag: vmaster-g845af298
SEP 29 - Refactor the JWA backend to utilize common code
https://github.com/kubeflow/kubeflow/commit/6eb007d0b7b37ece0eb9a072f12274aab0f8fe09
TAG PRESENT IN:
jupyter/notebook-controller/base/kustomization.yaml
- name: gcr.io/kubeflow-images-public/notebook-controller
  newName: gcr.io/kubeflow-images-public/notebook-controller
  newTag: vmaster-g6eb007d0
jupyter/notebook-controller/base_v3/kustomization.yaml
- name: gcr.io/kubeflow-images-public/notebook-controller
  newName: gcr.io/kubeflow-images-public/notebook-controller
  newTag: vmaster-g6eb007d0

**Comment by wg102** [comment]
The manifest branch 1.2.0 was released on November 20th.  

As it currently was on November 28th, The back end part – notebook-controller - was updated with the image https://github.com/kubeflow/kubeflow/commit/6eb007d0b7b37ece0eb9a072f12274aab0f8fe09 to refactor the JWA backend to utilize common code. On Nov 4th, an auto PR updated the notebook-controller tag to the one mentioned above, vmaster-g6eb007d0 (https://github.com/kubeflow/manifests/commit/0f673cf13ec9259bf06d7f84803c1722a3ffb33d).  

In the branch, the latest there was a PR to update some of the images in manifest two weeks ago, aka nov 16th https://github.com/kubeflow/manifests/pull/1645, but they updated with rather old tags. (See details for V1.2 in the comment above) 

- AUG 7 - Fix #5111: Trim whitespace from custom images 
- Sep 29 - Refactor the JWA backend to utilize common code (#5316) 

The front-end crud web app changes have not yet been implemented in the V1.2 branch since we couldn’t find the commit tag anywhere in the manifest. The latest tag is vmaster-g845af298 which has the commit of AUG 7 - Fix #5111: Trim whitespace from custom images (https://github.com/kubeflow/kubeflow/commit/845af298b733802e527562702cf283b6cc5b85cc)
As of now, there are still changes being made to both crud-web-app and jupyter-web-app in the kubeflow/kubeflow repository.

To summarize, v1.2 has the commit that contained the crud-web-app backend using common code, however maybe because it is still being worked on, the frontend for crud web apps has not been updated in the manifest. 
The commit to initialize the jupyter web app frontend is https://github.com/kubeflow/kubeflow/commit/89b13955b910d7320ffba6b03ed26ad4d1b56582 and was made on Oct 27 in the kubeflow/kubeflow repository but it’s commit tag is nowhere to be found in the manifest.
